### PR TITLE
Feature/cvsb 840

### PR DESCRIPTION
**Pull request by oalexandru**
- added API call to test-types in insertTestResult functionality so that it pulls a proper test_code and saves it together with the rest of the data for that specific test result
- mocked the API call in unit tests
- added another configuration named 'nonprod' where I specified the nonprod url for test-types microservice
NOTE:
* in order to run integration tests for test-results microservice you need to also have test-types microservice started(because it sends request to it)

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ custom:
       inMemory: true
       migrate: true
       seed: false
-      noStart: true
+      noStart: false
```
Comment (gmlazu) [inline serverless.yml]: The pipeline will fail if noStart is merged with `false`

File: src/models/TestResultsSchema.js [hunk 2]
```diff
@@ -52,8 +52,8 @@ const testTypesSchema = Joi.object().keys({
   numberOfSeatbeltsFitted: Joi.number().max(4).required(),
   lastSeatbeltInstallationCheckDate: Joi.date().required(),
   seatbeltInstallationCheckDate: Joi.boolean().required(),
-  testResult: Joi.any().only([ 'Failure', 'Pass', 'PRS', 'Abandonned', 'Successful', 'Unsuccessful' ]).required(),
-  prohibitionIssued: Joi.boolean().required(),
+  testResult: Joi.any().only([ 'failure', 'pass', 'prs', 'abandonned', 'successful', 'unsuccessful' ]).required(),
+  prohibitionIssued: Joi.any().only([ 'yes', 'no' ]).required(),
```
Comment (gmlazu) [inline src/models/TestResultsSchema.js]: As per spec, this has been changed to boolean. `Joi.boolean().required()`.

File: tests/resources/test-results.json [hunk 3]
```diff
@@ -1,79 +1,138 @@
 [
```
Comment (gmlazu) [inline tests/resources/test-results.json]: Please use the new mocks merged today into develop.

File: src/models/TestResultsSchema.js (hunk 2 above)
Comment (oalexandru) [reply src/models/TestResultsSchema.js]: fixed

File: tests/resources/test-results.json (hunk 3 above)
Comment (oalexandru) [reply tests/resources/test-results.json]: rebased and fixed

**Pull request closed without merging**